src/library/models.py

```diff
@@ -786,7 +807,7 @@ def update_id(self):
         so that client & server agree on one ID."""
         hl = self.highlight_object
         hl['id'] = self.pk
-        self.highlight_object = hl
+        highlight_object = hl
```

Comment:
This change seems unrelated to the purpose of the PR and potentially problematic.

Reply:
Good catch, @bgoldowsky. I've fixed this accidental change and run `python manage.py test` to make sure all tests pass.